server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -609,57 +704,45 @@ private static SnapshotsInRepo sortSnapshots(
         @Nullable String snapshotName,
         @Nullable String repoName
     ) {
-        final Predicate<SnapshotInfo> isAfter;
         switch (sortBy) {
             case START_TIME:
-                isAfter = filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
-                break;
+                return filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
             case NAME:
                 if (snapshotName == null) {
-                    assert repoName == null : "no snapshot name given but saw repo name [" + repoName + "]";
-                    isAfter = order == SortOrder.ASC
-                        ? snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) <= 0
-                        : snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) >= 0;
+                    // already covered by preflight filtering
+                    return null;
                 } else {
-                    isAfter = order == SortOrder.ASC
+                    // TODO: cover via pre-flight predicate
```

Comment:
No I was going for a follow-up to keep this one shorter.

Reply:
I should point out that the main reason adding this would blow up the size of the PR is a the peculiar behavior we have for `after` at the moment. The `after` parameter currently behaves differently from the `from_sort_value` in the count that it reports. It always reports the same overall count (the number of snapshots matching the query without `after`) while the `from_sort_value` simply reports the count of everything matching the request period.

I'm not sure the behavior of `after` is needed (as that param is just there for iteration anyway and I don't see how the total count even matters in use cases) but I think we should discuss + clean that up in a separate PR :)

Reply:
I suppose it makes sense to report back the count on every iteration/pagination, since this is a moving target. So you could see 51 first time for a page of 50 and then after pagination you may see 55. I think it would be nice to have the count in a UI be somewhat consistent with the last page of data shown?